browser/ui/tabs/brave_tab_strip_model.cc

```diff
@@ -0,0 +1,147 @@
+
+void BraveTabStripModel::SelectNextTab(UserGestureDetails detail) {
+  bool isMRUEnabled = Profile::FromBrowserContext(
+    GetActiveWebContents()->GetBrowserContext())
+      ->GetPrefs()
+      ->GetBoolean(kMRUCyclingEnabled);
+
+  if (isMRUEnabled) {
+     SelectTabMRU(false, detail);
+  } else {
+     SelectRelativeTab(true, detail);
+  }
+}
+
+void BraveTabStripModel::SelectPreviousTab(UserGestureDetails detail) {
+  bool isMRUEnabled = Profile::FromBrowserContext(
+    GetActiveWebContents()->GetBrowserContext())
+      ->GetPrefs()
+      ->GetBoolean(kMRUCyclingEnabled);
+
+  if (isMRUEnabled) {
+     SelectTabMRU(true, detail);
+  } else {
+     SelectRelativeTab(false, detail);
+  }
+}
+
+void BraveTabStripModel::SelectTabMRU(bool backward,
+                                      UserGestureDetails detail) {
+  if (current_mru_cycling_index == -1) {
+      // Start cycling
+
+      // Create a list of tab indexes sorted by time of last activation
+      for (int i = 0; i < count(); ++i) {
+        mru_cycle_list.push_back(i);
+      }
+
+      std::sort(mru_cycle_list.begin(),
+                mru_cycle_list.end(),
+                [this](int a, int b){
+        return GetWebContentsAt(a)->GetLastActiveTime() >
+          GetWebContentsAt(b)->GetLastActiveTime();
+        });
+
+      current_mru_cycling_index = 0;
+
+      // Create an event handler eating all keyboard events while tabing
+      ctrl_released_event_handler = std::make_unique<CtrlReleaseHandler>(this);
+
+      // Add the event handler
+      gfx::NativeWindow window = this->GetActiveWebContents()
+        ->GetTopLevelNativeWindow();
+#if defined(OS_MACOSX)
+      views::Widget::GetWidgetForNativeWindow(window)
+        ->GetRootView()
+        ->AddPreTargetHandler(ctrl_released_event_handler.get());
+#else
+      window->AddPreTargetHandler(ctrl_released_event_handler.get());
+#endif
+  }
+
+  int tabCount = mru_cycle_list.size();
+
+  if (backward) {
+    current_mru_cycling_index =
+      (current_mru_cycling_index - 1 % tabCount + tabCount) % tabCount;
+  } else {
+    current_mru_cycling_index =
+      (current_mru_cycling_index + 1 % tabCount + tabCount) % tabCount;
+  }
+
+  ActivateTabAt(mru_cycle_list[current_mru_cycling_index], detail);
+}
+
+void BraveTabStripModel::StopMRUCycling() {
+  current_mru_cycling_index = -1;
+  mru_cycle_list.clear();
+
+  if (ctrl_released_event_handler) {
+    // Remove the event handler
+    gfx::NativeWindow window = this->GetActiveWebContents()
+      ->GetTopLevelNativeWindow();
+#if defined(OS_MACOSX)
+    views::Widget::GetWidgetForNativeWindow(window)
+      ->GetRootView()
+      ->RemovePreTargetHandler(ctrl_released_event_handler.get());
+#else
+    window->RemovePreTargetHandler(ctrl_released_event_handler.get());
+#endif
+
+    ctrl_released_event_handler.reset();
+  }
+}
+
+
+BraveTabStripModel::CtrlReleaseHandler::CtrlReleaseHandler(
+  BraveTabStripModel* tab_strip): tab_strip(tab_strip) {}
```

Comment:
general advice: there is a formatting tool that plays well with Google style guide - `depot_tools/clang-format` - pls run it for all files.

Reply:
Done, I had only ran cpplint from the npm script of brave-browser.